File: boost-vertex-source-delivery/scripts/extract_case_study_figma_cards.py
from pathlib import Path
from PIL import Image

screenshot = Path('/home/ubuntu/screenshots/figma_2026-08-19_10-20-44_1172.webp')
output_dir = Path('/home/ubuntu/webdev-static-assets')
output_dir.mkdir(parents=True, exist_ok=True)

image = Image.open(screenshot).convert('RGB')
crops = [
    ('case-study-reading-network-figma.png', (34, 145, 301, 281)),
    ('case-study-reading-servers-figma.png', (317, 145, 586, 281)),
    ('case-study-reading-abstract-figma.png', (603, 145, 872, 281)),
]

for name, box in crops:
    image.crop(box).save(output_dir / name, 'PNG', optimize=True)

File: boost-vertex-source-delivery/scripts/extract_case_study_figma_cards.mjs
import { execFileSync } from 'node:child_process';
import { mkdirSync } from 'node:fs';

const screenshot = '/home/ubuntu/screenshots/figma_2026-08-19_10-20-44_1172.webp';
const outputDir = '/home/ubuntu/webdev-static-assets';
mkdirSync(outputDir, { recursive: true });

const crops = [
  ['case-study-reading-network-figma.png', '267x136+34+145'],
  ['case-study-reading-servers-figma.png', '269x136+317+145'],
  ['case-study-reading-abstract-figma.png', '269x136+603+145'],
];

for (const [name, geometry] of crops) {
  execFileSync('convert', [screenshot, '-crop', geometry, `${outputDir}/${name}`], { stdio: 'inherit' });
}
